docs/proposals/daemon-controller.md

```diff
@@ -0,0 +1,151 @@
+      spec:
+        node-selector: 
+          name: datastore-node
+        containers:
+          name: datastore-shard
+          image: kubernetes/sharded
+          ports:
+            - containerPort: 9042
+              name: main
+```
+  - commands that get info
+    - get (e.g. kubectl get dc)
+    - describe
+  - Modifiers
+    - delete
+    - stop: first we turn down the Daemon Controller foo, and then we turn down all pods matching the query “controller: foo”
+    - label
+    - update
+    - extension: rolling-update
+    - Daemon controllers will have labels, so you could, for example, list all daemon controllers with a certain label (the same way you would for a Replication Controller).
+  - In general, for all the supported features like get, describe, update, etc, the Daemon Controller will work in a similar way to the Replication Controller. However, note that the Daemon Controller and the Replication Controller are different constructs.
+
+### Health checks
+  - Ordinary health checks specified in the pod template will of course work to keep pods created by a Daemon Controller running.
+
+### Cluster Mutations
+  - When a new node is added to the cluster the daemon controller should start the daemon on the node (if the node’s labels match the user-specified selectors). This is a big advantage of the Daemon Controller compared to alternative ways of launching daemons and configuring clusters.
+  - Suppose the user launches a daemon controller that runs a logging daemon on all nodes labeled “tolog”. If the user then adds the “tolog” label to a node (that did not initially have the “tolog” label), the logging daemon should be launched on the node. Additionally, if a user removes the “tolog” label from a node, the logging daemon on that node should be killed.
+
+## Alternatives Considered
+
+An alternative way to launch daemons is to avoid going through the API server, and instead provide ways to package the daemon into the node. For example, users could:
+
+1. Include the daemon in the machine image
+2. Use config files to launch daemons
+3. Use static pod manifests to launch daemon pods when the node initializes
+
+These alternatives don’t work as well because the daemons won’t be well integrated into Kubernetes. In particular,
+
+1. In alternatives (1) and (2), health checking for the daemons would need to be re-implemented, or would not exist at all (because the daemons are not run inside pods). In the current proposal, the Kubelet will health-check daemon pods and restart them if necessary.
+2. In alternatives (1) and (2), binding services to a group of daemons is difficult (which is needed in use cases such as the sharded data store use case described above), because the daemons are not run inside pods
+3. A big disadvantage of these methods is that adding new daemons in existing nodes is difficult (for example, if a cluster manager wants to add a logging daemon after a cluster has been deployed).
```

Comment:
I think elaborating on this is important. Most of the world uses CM distributed /etc/init scrpits + sysv init/upstart/systemd to start daemons on nodes. Why is this inadequate? What can/can't you achive with this? Otherwise it sounds like you're writing a new daemon controller because you don't like salt. 

If your answer is: this puts daemons in pods, and pods have all these advantages, the next question is: why *not* just static pods (still not seeing a deal breaker, we upgraded the etcd binary with static pods just fine)? Suggest mentioning the flexibility a DCManager will give us in terms of things mentioned in other parts of this doc (eg: deployments, rollingupdate etc).